src/components/power-sensor/power-sensor.ts

```diff
@@ -0,0 +1,18 @@
+import type { StructType } from '../../types';
+import type { Sensor } from '../sensor';
+
+export type PowerSensorReadings = {
+  voltage?: number;
+  current?: number;
+  power?: number;
+};
+
+/** Represents any sensor that reports voltage, current, and/or power */
+export interface PowerSensor extends Sensor {
+  /** Get Voltage in volts */
+  getVoltage(extra?: StructType): Promise<readonly [number, boolean]>;
+  /** Get Current in amps */
+  getCurrent(extra?: StructType): Promise<readonly [number, boolean]>;
```

Comment:
we should document that getVoltage and getCurrent return whether the current is AC or not as the second return value